helios-testing/src/main/java/com/spotify/helios/testing/HeliosSoloDeployment.java

```diff
@@ -112,20 +111,36 @@
             .build();
   }
 
-  @Override
-  public HostAndPort address() {
-      return deploymentAddress;
-  }
+  private DockerHost containerDockerHost(final Builder builder) {
+    if (builder.containerDockerHost != null) {
+      return containerDockerHost;
+    }
 
-  private DockerHost containerDockerHostFromEnv() {
     if (isBoot2Docker(dockerInfo())) {
       return DockerHost.from(DefaultDockerClient.DEFAULT_UNIX_ENDPOINT, null);
-    } else {
-      return DockerHost.fromEnv();
     }
+
+    // otherwise construct a DockerHost from environment variables, *unless* DOCKER_HOST is set to
+    // localhost or 127.0.0.1 - which will never work inside a container. For those cases, we
+    // override the settings and use the unix socket instead.
+    final DockerHost dockerHost = DockerHost.fromEnv();
+    if (dockerHost.address().equals("localhost") || dockerHost.address().equals("127.0.0.1")) {
+      final String endpoint = DockerHost.DEFAULT_UNIX_ENDPOINT;
```

Comment:
I would only do this if the current OS is Linux, since otherwise I think there will be an exception down the road when trying to use the Unix sockets wrapper. If it's not Linux, we should just fail early.